### ISSUE TITLE
 Bidirectional ManyToOne: Issue with SelfAssociation

### ISSUE THREAD

**Issue by mevivs** [Impetus/kundera#293]
Entity:

```
@Table(name = "groups", schema = "KunderaExamples@secIdxCassandraTest")
@IndexCollection(columns = {@Index(name = "resourceId")})
public class Group {

   @Id
   @Column(name = "resourceId")
    private String resourceId;

@ManyToOne(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
@JoinColumn(name = "resourceId")
private Group parent;

@OneToMany(fetch = FetchType.EAGER, mappedBy = "parent")
    private List<Group> children;

@Column(name= "resourceName")
   private String resourceName;
```

Failing for fetching an entity object having parent object but no children.

Error produced is:

```
Caused by: java.lang.IllegalArgumentException: Can not set java.util.List field com.test.Group.children to com.test.Group
at sun.reflect.UnsafeFieldAccessorImpl.throwSetIllegalArgumentException(UnsafeFieldAccessorImpl.java:164)
at sun.reflect.UnsafeFieldAccessorImpl.throwSetIllegalArgumentException(UnsafeFieldAccessorImpl.java:168)
at sun.reflect.UnsafeObjectFieldAccessorImpl.set(UnsafeObjectFieldAccessorImpl.java:81)
at java.lang.reflect.Field.set(Field.java:680)
at com.impetus.kundera.property.PropertyAccessorHelper.set(PropertyAccessorHelper.java:100)
... 46 more

```

-Vivek

**Comment by joaonlima** [comment]
Do you think this will be on 2.6 ?

**Comment by mevivs** [comment]
Yes. I am working on it. BTW 2.6 will be releasing in next 10 days.

-Vivek

**Comment by mevivs** [comment]
A Fix has been added for this. Please verify with latest trunk. Will be releasing with latest trunk.

-Vivek

**Comment by joaonlima** [comment]
Thanks!
Unfortunately, I'm unable to give it a try for a while, sorry. But whenever possible I will try and post the results!

**Comment by mevivs** [comment]
Fixed and releasing with 2.6. Please verify.
-Vivek
